comp/core/tagger/impl-remote/remote.go

```diff
@@ -269,65 +272,78 @@ func (t *remoteTagger) GenerateContainerIDFromOriginInfo(originInfo origindetect
 		}
 	}()
 
-	expBackoff := backoff.NewExponentialBackOff()
-	expBackoff.InitialInterval = 500 * time.Millisecond
-	expBackoff.MaxInterval = 1 * time.Second
-	expBackoff.MaxElapsedTime = 15 * time.Second
-
-	var containerID string
+	// Generate cache key
+	initPrefix := ""
+	if originInfo.ExternalData.Init {
+		initPrefix = "i/"
+	}
+	key := cache.BuildAgentKey(
+		"remoteTagger",
+		"container_id",
+		initPrefix+originInfo.ExternalData.PodUID+"/"+originInfo.ExternalData.ContainerName,
+	)
 
-	err := backoff.Retry(func() error {
-		select {
-		case <-t.ctx.Done():
-			return &backoff.PermanentError{Err: errTaggerFailedGenerateContainerIDFromOriginInfo}
-		default:
-		}
+	cachedContainerID, err := cache.GetWithExpiration(key, func() (containerID string, err error) {
```

Comment:
for code clarity we define the function outside the call of `cache.GetWithExpiration()` 
I feel that we have to many nesting here.

Reply:
Sadly this does not seem possible without modifying the `pkg/util/cache/cache`, which I would prefer avoid doing 👍

Reply:
We checked together, all good 👍